bigquery/src/main/scala/com/yahoo/maha/executor/bigquery/BigqueryQueryExecutor.scala

```diff
@@ -0,0 +1,227 @@
+        .build()
+
+      val mHttpTransport = new ApacheHttpTransport(mHttpClient)
+      () => mHttpTransport
+    } finally {
+      fileInputStream.close()
+    }
+  }
+}
+
+class BigqueryQueryExecutor(
+  config: BigqueryQueryExecutorConfig,
+  lifecycleListener: ExecutionLifecycleListener
+) extends QueryExecutor
+  with Logging {
+
+  val engine: Engine = BigqueryEngine
+  val bigqueryClient = config.buildBigqueryClient()
+
+  def execute[T <: RowList](
+    query: Query,
+    rowList: T,
+    queryAttributes: QueryAttributes
+  ): QueryResult[T] = {
+    val acquiredQueryAttributes = lifecycleListener.acquired(query, queryAttributes)
+    val requestModel = query.queryContext.requestModel
+    val debugEnabled = requestModel.isDebugEnabled
+
+    if (!acceptEngine(query.engine)) {
+      throw new UnsupportedOperationException(
+        s"BigqueryQueryExecutor does not support query with engine=${query.engine}"
+      )
+    }
+
+    val formattedQuery = SqlFormatter.format(
+      query.asString,
+      FormatConfig(dialect = SqlDialect.StandardSQL)
+    )
+
+    if (debugEnabled) {
+      info(s"Running query : $formattedQuery")
+    }
+
+    val queryJobConfig = QueryJobConfiguration
+      .newBuilder(formattedQuery)
+      .setUseLegacySql(false)
+      .build()
+
+    val requestId = requestModel
+      .additionalParameters
+      .getOrElse(Parameter.RequestId, RequestIdValue(UUID.randomUUID().toString))
+      .asInstanceOf[RequestIdValue]
+      .value
+
+    val jobId = JobId.of(requestId)
+    val jobInfo = JobInfo.newBuilder(queryJobConfig).setJobId(jobId).build
+    val queryJob = bigqueryClient.create(jobInfo).waitFor()
+    val aliasColumnMap = query.aliasColumnMap
+    val queryColumns = query.asInstanceOf[BigqueryQuery].columnHeaders
+
+    rowList match {
+      case rl if rl.isInstanceOf[QueryRowList] =>
+        val queryRowList = rl.asInstanceOf[QueryRowList]
+
+        if (queryJob == null) {
+          error(s"Failed query : $formattedQuery")
+          val exception = new RuntimeException("BigQuery Job doesn't exist: " + jobId)
+          Try(lifecycleListener.failed(query, acquiredQueryAttributes, exception))
+          QueryResult(rl, acquiredQueryAttributes, QueryResultStatus.FAILURE, Option(exception))
+        } else if (queryJob.getStatus.getError() != null) {
+          error(s"Failed query : $formattedQuery")
+          val exception = new RuntimeException("BigQuery Job Failed: " + jobId)
+          Try(lifecycleListener.failed(query, acquiredQueryAttributes, exception))
+          QueryResult(rl, acquiredQueryAttributes, QueryResultStatus.FAILURE, Option(exception))
+        } else {
+          val result = queryJob.getQueryResults()
+          val resultColumns = result.getSchema().getFields()
+          if (resultColumns.size != queryColumns.size)
+            throw new RuntimeException("Mismatch in number of query and result columns")
+
+          val rowCount = result.getTotalRows()
```

Comment:
you could move this to inside the if statement below since it's only used there.

Reply:
Done